Sources/Filters/General/ShrinkPolyData/index.js

```diff
@@ -0,0 +1,338 @@
+          const shrunk = shrinkTowardsPoint(p, center, shrinkFactor);
+          newPointsData[outCount * 3] = shrunk[0];
+          newPointsData[outCount * 3 + 1] = shrunk[1];
+          newPointsData[outCount * 3 + 2] = shrunk[2];
+          newPolyData.push(outCount);
+          outCount++;
+        }
+        i += npts + 1;
+      }
+
+      newPolys.setData(new Uint32Array(newPolyData));
+    }
+
+    // Process triangle strips (convert to triangles and shrink)
+    if (inStrips) {
+      const stripData = inStrips.getData();
+      const newPolyData = [];
+
+      for (let i = 0; i < stripData.length; ) {
+        const npts = stripData[i];
+
+        for (let j = 0; j < npts - 2; j++) {
+          const ptId1 = stripData[i + j + 1];
+          const ptId2 = stripData[i + j + 2];
+          const ptId3 = stripData[i + j + 3];
+
+          // Compute centroid for triangle
+          vtkPolygon.computeCentroid([ptId1, ptId2, ptId3], inPoints, center);
+
+          // Shrink each point
+          const newIds = [];
+          const pts = [ptId1, ptId2, ptId3];
+          for (let k = 0; k < 3; k++) {
+            const ptId = pts[k];
+            const p = [
+              inPts[ptId * 3],
+              inPts[ptId * 3 + 1],
+              inPts[ptId * 3 + 2],
+            ];
+            const shrunk = shrinkTowardsPoint(p, center, shrinkFactor);
+            newPointsData[outCount * 3] = shrunk[0];
+            newPointsData[outCount * 3 + 1] = shrunk[1];
+            newPointsData[outCount * 3 + 2] = shrunk[2];
+            newIds.push(outCount);
+            outCount++;
+          }
+
+          // Reverse order for every other triangle to maintain proper orientation
+          if (j % 2) {
+            const tmp = newIds[0];
+            newIds[0] = newIds[2];
+            newIds[2] = tmp;
+          }
+
+          newPolyData.push(3, newIds[0], newIds[1], newIds[2]);
+        }
+        i += npts + 1;
+      }
+
+      if (newPolyData.length > 0) {
+        const existingPolyData = newPolys.getData();
+        const combinedPolyData = new Uint32Array(
+          existingPolyData.length + newPolyData.length
+        );
+        combinedPolyData.set(existingPolyData);
+        combinedPolyData.set(newPolyData, existingPolyData.length);
+        newPolys.setData(combinedPolyData);
+      }
+    }
+
+    // Set output
+    output.setPoints(newPoints);
+    output.setVerts(newVerts);
+    output.setLines(newLines);
+    output.setPolys(newPolys);
+
+    // Copy cell data
+    output.getCellData().passData(input.getCellData());
+  }
+
+  /**
+   * Shrink two points towards their midpoint by a shrink factor.
+   * @param {Vector3} p1 - The [x, y, z] coordinates of the first point
+   * @param {Vector3} p2 - The [x, y, z] coordinates of the second point
+   * @param {number} shrinkFactor - The shrink factor (0.0 to 1.0)
+   * @returns {Vector3[]} Array containing the two new points
+   */
+  publicAPI.shrinkLine = (p1, p2, shrinkFactor) => {
```

Comment:
pass returned output as optional parameter to avoid creating a new array each time